Modularization and adding of rule links

hcschild opened the pull request.

hcschild committed:
modularisation and links to jar,ipg,mtr and cr

hcschild committed:
missed cards module

bra1n commented on aggregators/mtg_cards/cards.js:
```diff
@@ -0,0 +1,38 @@
+function MtgCardLoader(){
+    var request = require("request");
```
`require` should happen on initialisation of the file, not inside a class constructor.

bra1n commented on aggregators/mtg_cards/cards.js:
```diff
@@ -0,0 +1,38 @@
+}
+MtgCardLoader.prototype.getContent = function(parameter,callback){this.find(parameter,callback);};
+
+module.exports = MtgCardLoader;
```
Why not use a real `class` instead of prototypes here?

``` js
export default class MtgCardLoader {
  constructor() {
    this.cardApi = "https://api.magicthegathering.io/v1/cards?name=";
  }
}
```

hcschild replied:
Because till now I didn't know about classes in ECMAScript . I did a read up on it and changed the code. :)

bra1n commented on aggregators/rules/cr.js:
```diff
@@ -0,0 +1,16 @@
+function CR(){
```
Should be a class, see above. :)

bra1n commented on server.js:
```diff
@@ -1,32 +1,57 @@
-    }
+	if(bot.user.id === msg.author.id) {return;}
+	var query = msg.content.split(" ");
+	if(!query){return;}
```
`!query` will never be `true`, because `query` will always be an Array with at least 1 element.

bra1n commented on server.js:
```diff
@@ -1,32 +1,57 @@
+	if(query.length > 1) {
+		parameter = query.slice(1).join(" ");
+	}else{
+		parameter = "";
```
Default `parameter` to `""` (Line 24), then you can drop the `else`.

hcschild replied:
I changed `parameter` to an `inline if` and made it a `const`. This should work too?

bra1n replied:
Sure, it's a bit harder to understand than my proposal, but fine with me.

bra1n reviewed (changes requested):
Good restructuring, needs some small work in the details.

hcschild committed:
requested changes for pull request "Modularization and adding of rule links"
require is no longer in the class constructor
changed aggregators to real classes
removed unnecessary if
changed if else to inline if
additionally changed the variables query, command and parameter to const

bra1n reviewed (approved):
Looking good!

bra1n commented on aggregators/mtg_cards/cards.js:
```diff
@@ -0,0 +1,38 @@
+}
+MtgCardLoader.prototype.getContent = function(parameter,callback){this.find(parameter,callback);};
+
+module.exports = MtgCardLoader;
```
👍
